ttnn/cpp/ttnn/operations/uniform/device/uniform_program_factory.cpp

```diff
@@ -0,0 +1,147 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "common/constants.hpp"
+#include "common/core_coord.h"
+#include "common/tt_backend_api_types.hpp"
+#include "tt_metal/common/work_split.hpp"
+#include "ttnn/tensor/types.hpp"
+#include "uniform_device_operation.hpp"
+
+namespace ttnn::operations::uniform {
+
+using namespace tt;
+using namespace tt::tt_metal;
+
+std::mt19937 rng(std::time(nullptr));
+std::uniform_int_distribution distribution(1, 1 << 20);
```

Comment:
why 1 << 20 and not the full range of uint ?

Reply:
Initially, I used `rand_uint_tile_init(i * seed)` with `i` is the tile_id. `1 << 20` used to avoid seed overflow. However, as I've updated the code to init only once, `1 << 20` is updated to max_int (`distribution` required 2nd param to be int).